jdbc/spring-cloud-starter-stream-source-jdbc/src/main/resources/META-INF/spring-configuration-metadata-whitelist.properties

```diff
@@ -1,3 +1,4 @@
 configuration-properties.classes=org.springframework.cloud.stream.app.jdbc.source.JdbcSourceProperties, \
-  org.springframework.cloud.stream.app.trigger.TriggerPropertiesMaxMessagesDefaultOne
+  org.springframework.cloud.stream.app.trigger.TriggerPropertiesMaxMessagesDefaultOne, \
+  org.springframework.boot.autoconfigure.jdbc.DataSourceProperties
```

Comment:
> Should the jdbc-sink have the similar DataSourceProperties?
Yep, certainly, oversight on my part.

And regarding your first comment, which I'm not sure I understand, Artem explained the motivation pretty clearly